### ISSUE TITLE
Test failure when attempting to run tests on master

### ISSUE THREAD

**Issue by kevinburkeotto** [gruntwork-io/terragrunt#342]
I get the following failure when I run tests on master:

```
[terragrunt] [fixture-download/local-relative] 2017/11/02 11:25:51 Running command: terraform --version
[terragrunt] 2017/11/02 11:25:51 Reading Terragrunt config file at fixture-download/local-relative/terraform.tfvars
[terragrunt] 2017/11/02 11:25:51 Cleaning up existing *.tf files in /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt/_bTmU0G7yWqwUBGZ8iebosw-KyQ/6fc0zh5Br1YZ0V1jIxjEhrFZIrs
[terragrunt] 2017/11/02 11:25:51 Downloading Terraform configurations from file:///Users/kevin/src/github.com/gruntwork-io/terragrunt/test/fixture-download into /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt/_bTmU0G7yWqwUBGZ8iebosw-KyQ/6fc0zh5Br1YZ0V1jIxjEhrFZIrs using terraform init
[terragrunt] [fixture-download/local-relative] 2017/11/02 11:25:51 Running command: terraform init -from-module=file:///Users/kevin/src/github.com/gruntwork-io/terragrunt/test/fixture-download /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt/_bTmU0G7yWqwUBGZ8iebosw-KyQ/6fc0zh5Br1YZ0V1jIxjEhrFZIrs
Copying configuration from "file:///Users/kevin/src/github.com/gruntwork-io/terragrunt/test/fixture-download"...
Terraform initialized in an empty directory!

The directory has no Terraform configuration files. You may begin working
with Terraform immediately by creating Terraform configuration files.
[terragrunt] 2017/11/02 11:25:51 Copying files from fixture-download/local-relative into /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt/_bTmU0G7yWqwUBGZ8iebosw-KyQ/6fc0zh5Br1YZ0V1jIxjEhrFZIrs/relative
[terragrunt] 2017/11/02 11:25:51 Setting working directory to /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt/_bTmU0G7yWqwUBGZ8iebosw-KyQ/6fc0zh5Br1YZ0V1jIxjEhrFZIrs/relative
[terragrunt] 2017/11/02 11:25:51 Running command: terraform apply

Error: Failed to load root config module: Error loading modules: module remote: not found, may need to be downloaded using 'terraform get'


--- FAIL: TestLocalDownloadWithRelativePath (0.11s)
	integration_test.go:515: Failed to run Terragrunt command 'terragrunt apply --terragrunt-non-interactive --terragrunt-working-dir fixture-download/local-relative' due to error: exit status 1
FAIL
FAIL	github.com/gruntwork-io/terragrunt/test	0.125s
```

You can reproduce (hopefully) by running `$ aws-vault exec testing -- go test -run TestLocalDownloadWithRelativePath ./...` ("testing" is an isolated AWS profile, aws-vault is here: github.com/99designs/aws-vault). This is the only test failure.

I'm running Terraform v0.10.8, and Terragrunt commit 967e9eebf669d5c19aad4bbd68c4497c099f0e8b, on a Mac. I'm running Go tip, but I doubt the Go version matters based on the build output.

**Comment by kevinburkeotto** [comment]
Note that test also fails when I run the whole suite, it's just faster to see the failure when the test is run in isolation.

**Comment by kevinburkeotto** [comment]
Here's the output when I run the tests with TF_LOG=debug enabled.

```
[terragrunt] [fixture-download/local-relative] 2017/11/02 11:32:00 Running command: terraform --version
[terragrunt] 2017/11/02 11:32:00 Reading Terragrunt config file at fixture-download/local-relative/terraform.tfvars
[terragrunt] 2017/11/02 11:32:00 Cleaning up existing *.tf files in /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt/_bTmU0G7yWqwUBGZ8iebosw-KyQ/6fc0zh5Br1YZ0V1jIxjEhrFZIrs
[terragrunt] 2017/11/02 11:32:00 Downloading Terraform configurations from file:///Users/kevin/src/github.com/gruntwork-io/terragrunt/test/fixture-download into /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt/_bTmU0G7yWqwUBGZ8iebosw-KyQ/6fc0zh5Br1YZ0V1jIxjEhrFZIrs using terraform init
[terragrunt] [fixture-download/local-relative] 2017/11/02 11:32:00 Running command: terraform init -from-module=file:///Users/kevin/src/github.com/gruntwork-io/terragrunt/test/fixture-download /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt/_bTmU0G7yWqwUBGZ8iebosw-KyQ/6fc0zh5Br1YZ0V1jIxjEhrFZIrs
2017/11/02 11:32:00 [INFO] Terraform version: 0.10.8
2017/11/02 11:32:00 [INFO] Go runtime version: go1.9.1
2017/11/02 11:32:00 [INFO] CLI args: []string{"/usr/local/Cellar/terraform/0.10.8/bin/terraform", "init", "-from-module=file:///Users/kevin/src/github.com/gruntwork-io/terragrunt/test/fixture-download", "/var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt/_bTmU0G7yWqwUBGZ8iebosw-KyQ/6fc0zh5Br1YZ0V1jIxjEhrFZIrs"}
2017/11/02 11:32:00 [DEBUG] Attempting to open CLI config file: /Users/kevin/.terraformrc
2017/11/02 11:32:00 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2017/11/02 11:32:00 [DEBUG] CLI config is &main.Config{Providers:map[string]string{}, Provisioners:map[string]string{}, DisableCheckpoint:false, DisableCheckpointSignature:false, PluginCacheDir:"", Credentials:map[string]map[string]interface {}(nil), CredentialsHelpers:map[string]*main.ConfigCredentialsHelper(nil)}
2017/11/02 11:32:00 [INFO] CLI command args: []string{"init", "-from-module=file:///Users/kevin/src/github.com/gruntwork-io/terragrunt/test/fixture-download", "/var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt/_bTmU0G7yWqwUBGZ8iebosw-KyQ/6fc0zh5Br1YZ0V1jIxjEhrFZIrs"}
Copying configuration from "file:///Users/kevin/src/github.com/gruntwork-io/terragrunt/test/fixture-download"...
2017/11/02 11:32:00 [DEBUG] plugin: waiting for all plugin processes to complete...
Terraform initialized in an empty directory!

The directory has no Terraform configuration files. You may begin working
with Terraform immediately by creating Terraform configuration files.
[terragrunt] 2017/11/02 11:32:00 Copying files from fixture-download/local-relative into /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt/_bTmU0G7yWqwUBGZ8iebosw-KyQ/6fc0zh5Br1YZ0V1jIxjEhrFZIrs/relative
[terragrunt] 2017/11/02 11:32:00 Setting working directory to /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt/_bTmU0G7yWqwUBGZ8iebosw-KyQ/6fc0zh5Br1YZ0V1jIxjEhrFZIrs/relative
[terragrunt] 2017/11/02 11:32:00 Running command: terraform apply
2017/11/02 11:32:00 [INFO] Terraform version: 0.10.8
2017/11/02 11:32:00 [INFO] Go runtime version: go1.9.1
2017/11/02 11:32:00 [INFO] CLI args: []string{"/usr/local/Cellar/terraform/0.10.8/bin/terraform", "apply"}
2017/11/02 11:32:00 [DEBUG] Attempting to open CLI config file: /Users/kevin/.terraformrc
2017/11/02 11:32:00 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2017/11/02 11:32:00 [DEBUG] CLI config is &main.Config{Providers:map[string]string{}, Provisioners:map[string]string{}, DisableCheckpoint:false, DisableCheckpointSignature:false, PluginCacheDir:"", Credentials:map[string]map[string]interface {}(nil), CredentialsHelpers:map[string]*main.ConfigCredentialsHelper(nil)}
2017/11/02 11:32:00 [INFO] CLI command args: []string{"apply"}
2017/11/02 11:32:00 [DEBUG] plugin: waiting for all plugin processes to complete...

Error: Failed to load root config module: Error loading modules: module remote: not found, may need to be downloaded using 'terraform get'


--- FAIL: TestLocalDownloadWithRelativePath (0.11s)
	integration_test.go:515: Failed to run Terragrunt command 'terragrunt apply --terragrunt-non-interactive --terragrunt-working-dir fixture-download/local-relative' due to error: exit status 1
FAIL
FAIL	github.com/gruntwork-io/terragrunt/test	0.130s
```

**Comment by kevinburkeotto** [comment]
Update, it passes if I do the following

```
pushd test/fixtures/local-relative
    terragrunt get 
    terragrunt init
popd
go test ./...
```

**Comment by brikis98** [comment]
That's weird. The tests just passed on CircleCI for that commit. They pass for me locally too:

```
$ go test -v -run TestLocalDownloadWithRelativePath $(glide novendor)

?   	github.com/gruntwork-io/terragrunt/aws_helper	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/gruntwork-io/terragrunt/cli	0.021s [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/gruntwork-io/terragrunt/config	0.021s [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/gruntwork-io/terragrunt/configstack	0.021s [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/gruntwork-io/terragrunt/dynamodb	0.015s [no tests to run]
?   	github.com/gruntwork-io/terragrunt/errors	[no test files]
?   	github.com/gruntwork-io/terragrunt/options	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/gruntwork-io/terragrunt/remote	0.021s [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/gruntwork-io/terragrunt/shell	0.019s [no tests to run]
=== RUN   TestLocalDownloadWithRelativePath
[terragrunt] [fixture-download/local-relative] 2017/11/02 23:21:02 Running command: terraform --version
[terragrunt] 2017/11/02 23:21:02 Reading Terragrunt config file at fixture-download/local-relative/terraform.tfvars
[terragrunt] 2017/11/02 23:21:02 Cleaning up existing *.tf files in /var/folders/39/w935myd14p9197wbp67pz4940000gn/T/terragrunt/8e0-cHX69r27X-3xaasiGNzTg-s/up-hUBVKu-INtjG7UXCniHCC2JQ
[terragrunt] 2017/11/02 23:21:02 Downloading Terraform configurations from file:///Users/brikis98/source/go/src/github.com/gruntwork-io/terragrunt/test/fixture-download into /var/folders/39/w935myd14p9197wbp67pz4940000gn/T/terragrunt/8e0-cHX69r27X-3xaasiGNzTg-s/up-hUBVKu-INtjG7UXCniHCC2JQ using terraform init
[terragrunt] [fixture-download/local-relative] 2017/11/02 23:21:02 Running command: terraform init -from-module=file:///Users/brikis98/source/go/src/github.com/gruntwork-io/terragrunt/test/fixture-download /var/folders/39/w935myd14p9197wbp67pz4940000gn/T/terragrunt/8e0-cHX69r27X-3xaasiGNzTg-s/up-hUBVKu-INtjG7UXCniHCC2JQ
Copying configuration from "file:///Users/brikis98/source/go/src/github.com/gruntwork-io/terragrunt/test/fixture-download"...
Terraform initialized in an empty directory!

The directory has no Terraform configuration files. You may begin working
with Terraform immediately by creating Terraform configuration files.
[terragrunt] 2017/11/02 23:21:02 Copying files from fixture-download/local-relative into /var/folders/39/w935myd14p9197wbp67pz4940000gn/T/terragrunt/8e0-cHX69r27X-3xaasiGNzTg-s/up-hUBVKu-INtjG7UXCniHCC2JQ/relative
[terragrunt] 2017/11/02 23:21:02 Setting working directory to /var/folders/39/w935myd14p9197wbp67pz4940000gn/T/terragrunt/8e0-cHX69r27X-3xaasiGNzTg-s/up-hUBVKu-INtjG7UXCniHCC2JQ/relative
[terragrunt] 2017/11/02 23:21:02 Running command: terraform apply
data.template_file.test: Refreshing state...
data.template_file.test: Refreshing state...

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

test = Hello, World
[terragrunt] [fixture-download/local-relative] 2017/11/02 23:21:02 Running command: terraform --version
[terragrunt] 2017/11/02 23:21:02 Reading Terragrunt config file at fixture-download/local-relative/terraform.tfvars
[terragrunt] 2017/11/02 23:21:02 Cleaning up existing *.tf files in /var/folders/39/w935myd14p9197wbp67pz4940000gn/T/terragrunt/8e0-cHX69r27X-3xaasiGNzTg-s/up-hUBVKu-INtjG7UXCniHCC2JQ
[terragrunt] 2017/11/02 23:21:02 Downloading Terraform configurations from file:///Users/brikis98/source/go/src/github.com/gruntwork-io/terragrunt/test/fixture-download into /var/folders/39/w935myd14p9197wbp67pz4940000gn/T/terragrunt/8e0-cHX69r27X-3xaasiGNzTg-s/up-hUBVKu-INtjG7UXCniHCC2JQ using terraform init
[terragrunt] [fixture-download/local-relative] 2017/11/02 23:21:02 Running command: terraform init -from-module=file:///Users/brikis98/source/go/src/github.com/gruntwork-io/terragrunt/test/fixture-download /var/folders/39/w935myd14p9197wbp67pz4940000gn/T/terragrunt/8e0-cHX69r27X-3xaasiGNzTg-s/up-hUBVKu-INtjG7UXCniHCC2JQ
Copying configuration from "file:///Users/brikis98/source/go/src/github.com/gruntwork-io/terragrunt/test/fixture-download"...
Terraform initialized in an empty directory!

The directory has no Terraform configuration files. You may begin working
with Terraform immediately by creating Terraform configuration files.
[terragrunt] 2017/11/02 23:21:02 Copying files from fixture-download/local-relative into /var/folders/39/w935myd14p9197wbp67pz4940000gn/T/terragrunt/8e0-cHX69r27X-3xaasiGNzTg-s/up-hUBVKu-INtjG7UXCniHCC2JQ/relative
[terragrunt] 2017/11/02 23:21:02 Setting working directory to /var/folders/39/w935myd14p9197wbp67pz4940000gn/T/terragrunt/8e0-cHX69r27X-3xaasiGNzTg-s/up-hUBVKu-INtjG7UXCniHCC2JQ/relative
[terragrunt] 2017/11/02 23:21:02 Running command: terraform apply
data.template_file.test: Refreshing state...
data.template_file.test: Refreshing state...

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

test = Hello, World
--- PASS: TestLocalDownloadWithRelativePath (0.88s)
PASS
ok  	github.com/gruntwork-io/terragrunt/test	0.895s
?   	github.com/gruntwork-io/terragrunt/test/helpers	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/gruntwork-io/terragrunt/util	0.020s [no tests to run]
?   	github.com/gruntwork-io/terragrunt	[no test files]
```

**Comment by brikis98** [comment]
Is it possible that this is a left over from your work on #338? If you delete `/var/folders/39/w935myd14p9197wbp67pz4940000gn/T/terragrunt`, and re-run, does the test pass?

**Comment by kevinburkeotto** [comment]
I just removed that folder and got this error on the next run:

```
--- FAIL: TestTerragruntStackCommands (22.81s)
	integration_test.go:525: Copying fixture-stack/ to /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt-test258424976
	integration_test.go:515: Failed to run Terragrunt command 'terragrunt apply-all --terragrunt-non-interactive --terragrunt-working-dir /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt-test258424976/fixture-stack/mgmt -var terraform_remote_state_s3_bucket="terragrunt-test-bucket-mc5xnh"' due to error: Encountered the following errors:
		NoSuchBucket: The specified bucket does not exist
			status code: 404, request id: BFCB34FFB7AA3F4C, host id: U8A8jQc+tp61Gu01LTxnp+eaw7nGA1ppU1FvTGcDG7+rqm1gwcRA9c1p5Sbq6qsR9UfOx8nEXjE=
		Cannot process module Module /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt-test258424976/fixture-stack/mgmt/bastion-host (dependencies: [/var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt-test258424976/fixture-stack/mgmt/vpc, /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt-test258424976/fixture-stack/mgmt/kms-master-key]) because one of its dependencies, Module /var/folders/sf/fsn3_vgd0n98r0jb86bgp83r0000gn/T/terragrunt-test258424976/fixture-stack/mgmt/vpc (dependencies: []), finished with an error: NoSuchBucket: The specified bucket does not exist
			status code: 404, request id: BFCB34FFB7AA3F4C, host id: U8A8jQc+tp61Gu01LTxnp+eaw7nGA1ppU1FvTGcDG7+rqm1gwcRA9c1p5Sbq6qsR9UfOx8nEXjE=
	integration_test.go:645: Deleting test s3 bucket terragrunt-test-bucket-mc5xnh
```

**Comment by brikis98** [comment]
Bizarre. Terragrunt tests have been running very consistently in [CircleCI](https://circleci.com/gh/gruntwork-io/terragrunt), my computer, and everyone else who has tried it for months...

That error is particularly odd. It's not obvious to me why the bucket wouldn't exist, unless you happened to hit a one in a billion chance where the pseudo-random bucket IDs we use happened to clash with another test that deleted the bucket. Is there more log output or a stack trace? And out of curiosity, does the test pass if you re-run it?

**Comment by brikis98** [comment]
I'm not able to repro this locally or in CircleCI. I did fix some test failures in #369, so perhaps this took care of it too? Closing for now, please re-open if this is still an issue.
